### ISSUE TITLE
Show and allow interaction with cell toolbar when its cell is hovered

### ISSUE THREAD

**Issue by firai** [jupyterlab/jupyterlab#13153]
### Problem

Using a given cell's toolbar currently requires first focusing on the cell by clicking on the cell. This feels a little clunky, since a number of operations in the toolbar aren't available in the right click menu, and since JupyterLab doesn't have the on-hover between-cell cell insertion toolbars and run buttons that exist other notebook interfaces (such as VSCode and Google Colab).

### Proposed Solution

Show a cell's toolbar when the cell is hovered, and allow interaction with the toolbar.

**Comment by firai** [comment]
@meeseeksdev tag "pkg:cell-toolbar"

**Comment by jweill-aws** [comment]
This is something that I considered when initially working on the cell toolbar. UI controls that appear _only_ on hover are not great for accessibility, so we should show the cell toolbar both on click/activate and on hover if possible. There's also an enhancement request to make the cell toolbar more visible even if it would overlap cell contents (#12223). Thank you for opening this issue!

**Comment by firai** [comment]
Keeping the cell toolbar on when a cell is focused/activated also makes sense from a regular usage point of view and for the purpose of not breaking current behavior and user expectations

**Comment by firai** [comment]
@jweill-aws Out of curiosity, does the current implementation of the cell toolbar assume that there would only be one instance at any given time (i.e. allowing both the active cell and the hovered cell to have toolbars would require significant restructuring), or do you think adding cell hover as another event that triggers instantiating a cell toolbar could be relatively simple to implement?

**Comment by jweill-aws** [comment]
@firai Yes, at present there is exactly one instance of the cell toolbar. This differs from the [enhanced cell toolbar](https://github.com/jupyterlab-contrib/jlab-enhanced-cell-toolbar), from which some of the cell toolbar's code derives. The enhanced cell toolbar added a toolbar atop _every_ cell, only making the active cell's toolbar visible. I found problems related to the enhanced cell toolbar when cells were added, deleted, or moved around, though. (This was a few months ago; I don't remember all the specifics.)
